<launch>
  <!-- Use bag time -->
  <param name="/use_sim_time" value="true"/>

  <!-- 1. odom -> base_link TF from /odometry -->
  <node name="odom_tf_broadcaster"
        pkg="second_project"
        type="odom_tf_broadcaster"
        output="log"/>

  <!-- 2. Merge front & rear lasers -->
  <node name="scan_merger"
        pkg="ira_laser_tools"
        type="laserscan_multi_merger"
        output="screen">
    <param name="scan_topic"        value="/scan_front,/scan_back"/>
    <param name="publish_topic"     value="/scan_raw"/>
    <param name="destination_frame" value="base_link"/>
    <param name="angle_min"         value="-3.14159"/>
    <param name="angle_max"         value=" 3.14159"/>
    <param name="range_min"         value="0.20"/>
  </node>

  <!-- 3. Range filter (optional but recommended) -->
  <node name="scan_filter"
        pkg="laser_filters"
        type="scan_to_scan_filter_chain"
        output="screen">
    <rosparam command="load" file="$(find second_project)/config/laser_filter.yaml"/>
    <remap from="scan"         to="/scan_raw"/>
    <remap from="scan_filtered" to="/scan"/>
  </node>

  <!-- 4. SLAM GMapping -->
  <node name="slam_gmapping"
        pkg="gmapping"
        type="slam_gmapping"
        output="screen">
    <param name="base_frame" value="base_link"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_frame"  value="map"/>
  </node>

  <!-- 5. RViz -->
  <node name="rviz"
        pkg="rviz"
        type="rviz"
        args="-d $(find second_project)/rviz/mapping.rviz"/>
</launch>
